typegen/header.d.ts

```diff
@@ -0,0 +1,270 @@
+import {
```

Comment:
Is there any way to verify that these are correct? I don't know typescript, so these look correct to me as far as I understand them, but is there an automated way to check them?

Reply:
The only automated way that came to my mind is just to try compile it with typescript compiler. Compiler will check definition file to some extent. However it only makes sens to compile final index.d.ts, because header.d.ts is just part of final file. It only includes some manual definitions that we can use later while iterating through prop-types

Reply:
I can include typescript compiler check of generated artifact like this:
```suggestion
"generate-types": "node ./typegen/generate-types.js && node ./typegen/generate-modules.js && tsc --noEmit --strict index.d.ts"
```
However this will require few additional devDependencies: 
"typescript" - compiler itself and "@types/react" - type definitions for React library